uqcsbot/scripts/wolfram.py

```diff
@@ -0,0 +1,192 @@
+    result = json.loads(http_response.content)['queryresult']
+    if not result['success'] or result["error"]:
+        bot.post_message(command.channel, "Please rephrase your query. Wolfram could not compute.")
+
+    subpods = get_subpods(result['pods'])
+    message = ""
+    for title, subpod in subpods:
+        plaintext = subpod["plaintext"]
+
+        # Prefer a plain text representation to the image
+        if plaintext != "" and plaintext != "* * * * * *":
+            message += f'{title}: {plaintext}\n'
+        else:
+            image_url = subpod['img']['src']
+            image_title = subpod['img']['title']
+            message += f'{image_title}:\n{image_url}\n' if image_title else f'{image_url}\n'
+
+    bot.post_message(command.channel, message)
+
+
+@bot.on_command("wolfram")
+async def handle_wolfram(command: Command):
+    """
+    This uses wolfram's conversation api to return a short response that can be replied to in a thread.
+    If the response cannot be replied to a general short answer response is displayed instead.
+
+    Example Usage:
+    !wolfram Solve Newton's Second Law for mass
+    !wolfram What is the distance from Earth to Mars?
+
+    and then start a thread to continue to conversation
+    """
+    api_url = r"http://api.wolframalpha.com/v1/conversation.jsp"
+    search_query = command.arg
+    http_response = await bot.run_async(requests.get, api_url, params={'input': search_query, 'appid': APP_ID})
+
+    # Check if the response is okay
+    if http_response.status_code != requests.codes.ok:
+        bot.post_message(command.channel, "There was a problem getting the response")
+        return
+
+    wolfram_result = json.loads(http_response.content)
+    if 'error' in wolfram_result:
+        error = wolfram_result['error']
+        if error == "No result is available":
+            # If no conversational result is available just return a normal short answer
+            short_response = await short_answer(search_query)
+            bot.post_message(command.channel, short_response)
+            return
+        else:
+            bot.post_message(command.channel, error)
+            return
+
+    # Unpack the response in order to construct the reply message
+    result = wolfram_result['result']  # This is the answer to our question
+    reply_host = wolfram_result['host']  # This is the hostname to ask the next question to
+    conversation_id = wolfram_result['conversationID']  # Used to continue the conversation
+    s_output = wolfram_result.get('s', None)  # s is only sometimes returned but is vital if it is returned
+
+    # TODO: Is there a better option than storing the id in the fallback?
+    # Here we store the conversation ID in the fallback so we can get it back later.
+    # We also store an identifier string to check against later and the reply_host and s_output string
+    attachments = [{
+        'fallback': f'WolframCanReply {reply_host} {s_output} {conversation_id}',
```

Comment:
is `s_output` is None, this will have a double space, is that an issue?

Reply:
also I'm happy to keep it for now, but I'm sure we can do something better than stuffing it in fallback hehe :^) maybe we can build another robot brain (essentially just an in-memory key-value store we used in our jsbot)